### ISSUE TITLE
Feature Request: ID Span Queries

### ISSUE THREAD

**Issue by Mythologos** [bdusell/qfunnel#8]
When performing hyperparameter searches or cross-validation experiments, it is possible that a whole set of experiments need to be altered or prioritized. However, for job deletion and for bumping job order, current syntax can be clunky or tedious for handling a lot of jobs. As a result, I would like to propose another way of querying the desired jobs which may make it easier to use `qfunnel` with a large number of experiments at once.

Currently, jobs are listed by their ID in `qf list` when they are running, and they are given a temporary batched ID beginning with 'x' before they start running. Such identifications can be used to our advantage. In particular, if a whole group of jobs is meant to be removed, then we could indicate such jobs by the start ID and the end ID. 

One way to implement this is to add `--start` and `--end` flags to each of `qf delete` and `qf bump`. We would then have commands like the following:

```
qf delete --start x001 --end x024
```

```
qf delete --start 000001 --end 000053
```

```
qf delete --start 000054 --end x010
```

```
qf bump --start x001 --end x024
```

Note that there are more cases for `delete` than for `bump`. The latter only makes sense on buffered jobs, but the former could be on any span of jobs. Thus, it's best to consider the ordering of the jobs relative to their number, with buffered IDs automatically being ordered after running IDs (as per `qf list`'s current order). Moreover, both uses of span-based selection would need to handle gaps in the value (*i.e.*, say that a job was already deleted in `qf bump` between x001 and x024, or that other people submitted jobs which received IDs between 000001 and 000053).

Questions also arise about how and whether this syntax (or a variant of it) would interact with other versions of deletion and bumping. It's possible that there's a meaningful combination, but they could also be used to the exclusion of others.

**Comment by bdusell** [comment]
FYI, in bash, you can generate a range of values with

```bash
qf delete x{001..024}
qf delete {000001..000053}
qf bump {001..024}
```

But I think what you're asking for is a way to select a range of jobs according to the output of `qf list`, right? So it would skip any numbers that aren't jobs managed by qfunnel.

**Comment by Mythologos** [comment]
Yes, that's right. (And thanks for the tip about bash!) As I understand it, `delete` allows for regex-based selections for titles, but this seems to be a more difficult option in cases where larger numbers of jobs are submitted. In my personal experience thus far, it's often that an issue or need concerns a whole group of jobs rather than a potentially scattered selection of them.
